vars/branchRepo.groovy

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/groovy
+
+def call(String gitRepoUrl, String gitRef, String checkoutDir, String branchName, Closure body) {
+    checkoutGitRepo {
+        repoUrl = gitRepoUrl
+        branch = gitRef
+        targetDir = checkoutDir
+    }
+    dir(checkoutDir) {
+        sshagent(['jenkinsgithub']) {
```

Comment:
Would it be worthwhile to make the credential name here, and the git user.name and email below, configurable?